packages/nimble-blazor/Examples/Demo.Shared/Pages/ComponentsDemo.razor

```diff
@@ -177,6 +177,11 @@
                 <span slot="checked-message">On</span>
             </NimbleSwitch>
         </div>
+        <div class="sub-container table-container">
+            <div class="container-label">Table</div>
+            <NimbleTable @bind-Data="TableData"></NimbleTable>
+            <NimbleButton @onclick=@(_ => UpdateTableData(10000))>Update Data</NimbleButton>
```

Comment:
When I click the Update Data button in the [build associated with this PR](https://60e89457a987cf003efc0a5b-ryrzrvteep.chromatic.com/blazor-client-app/wwwroot/) the page hits 100% CPU and stops responding to interaction (tested in Firefox if that matters). Is this expected for a data set this big or is something going wrong?

Reply:
Hmmmm...this is concerning. How do I see the example for the most recent build ('Fix docs')? Needless to say, this is not expected, and something is definitely going wrong.

Reply:
I get to it by clicking the storybook link in the build status and then clicking the blazor example link on the storybook landing page.

Reply:
<img width="950" alt="image" src="https://user-images.githubusercontent.com/10500124/208548048-0e9ae5a6-a70b-4191-97e8-e752868d53c7.png">

<img width="900" alt="image" src="https://user-images.githubusercontent.com/10500124/208548126-5e92d6d8-66fb-49ae-acab-3a514adfcaa0.png">

Reply:
So, my debugging efforts have revealed some interesting, unfortunate behaviors for Blazor (WebAssembly specifically). First, I'll point out that the linked example app is for a Blazor WebAssembly app (which I believe is necessary to host off of Chromatic). The difference in performance between Blazor Server and Blazor WebAssembly is staggering, and my opinion is that we shouldn't spend a lot of effort (for now) investigating solutions to it, as we have no known clients requiring or planning on using Blazor WebAssembly.

Using some of the debugging changes I made here is the performance profile I noticed (wall clock times). Notably, I am allocating 10K array elements in the constructor for the demo page:

Load time (allocate 10K elements):
- Blazor Server: Instant
- Blazor WebAssembly: ~25 seconds

Set Table (to 10K rows):
- Blazor Server: < 1 second
- Blazor WebAssembly: ~5 seconds

For now. I will make a change that will update the Table with far less data, but such that a user will still be able to scroll through a sizeable set of data (100 rows).

Reply:
It was the Faker library causing the problems. I've also tried the [Bogus](https://github.com/bchavez/Bogus) library, and it yields similar performance problems on init. For now, I'll simply reduce the total number of rows we're allocating on update.

Note that this is only an issue in Blazor WebAssembly. Blazor Server is quite performant with 10K rows.